src/scripts/main.js

```diff
@@ -1,3 +1,66 @@
 'use strict';
 
-// write code here
+const appendRowButton = document.querySelector('.append-row');
+const removeRowButton = document.querySelector('.remove-row');
+const appendColumnButton = document.querySelector('.append-column');
+const removeColumnButton = document.querySelector('.remove-column');
+
+const table = document.querySelector('.field').tBodies[0];
+
+const maxLength = 10;
+const minLength = 2;
+
+function updateButtonState() {
+  const rowsCount = table.rows.length;
+  const columnsCount = table.rows[0].cells.length;
+
+  appendRowButton.disabled = rowsCount >= maxLength;
+  removeRowButton.disabled = rowsCount <= minLength;
+  appendColumnButton.disabled = columnsCount >= maxLength;
+  removeColumnButton.disabled = columnsCount <= minLength;
+}
```

Comment:
Good approach to use the special function, but on row changing action only row buttons can change their state. So no need to update the column buttons state. Same for columns